src/Poll.tsx

```diff
@@ -343,13 +343,13 @@ function Poll<TData = any, TError = any, TQueryParams = { [key: string]: any }>(
   // Compose Contexts to allow for URL nesting
   return (
     <RestfulReactConsumer>
-      {contextProps => {
+      {async contextProps => {
```

Comment:
The poll component needs some kind of rework to support this I think.

Placing this async here is what creates all the invariant errors we see in the tests - React tries to render the Promise object of this function

I'm unsure how you want to handle this?